deploy/backup/copy-backup.sh

```diff
@@ -68,58 +76,90 @@ enable_logging() {
     fi
 }
 
-check_input() {
-    local backup_dest=$1
-    local dest_dir=$2
+check_input_namespace() {
+    local namespace=${1}
 
-    echo
-    if [ -z "$backup_dest" ] || [ -z "$dest_dir" ]; then
-        usage
+    if [ -n "$namespace" ]; then
+        ctrl="$ctrl -n $namespace"
     fi
+}
 
-    if [ ! -e "$dest_dir" ]; then
-        mkdir -p "$dest_dir"
-    fi
+check_input_destination() {
+  local backup_dest=$1
+  local dest_dir=$2
+
+  if [ -z "$backup_dest" ] || [ -z "$dest_dir" ]; then
+      usage
+  fi
+
+  if [ ! -e "$dest_dir" ]; then
+      mkdir -p "$dest_dir"
+  fi
+
+  if [ "${backup_dest:0:4}" = "pvc/" ]; then
+      if ! $ctrl get "$backup_dest" 1>/dev/null; then
+          printf "[ERROR] '%s' PVC doesn't exists.\n\n" "$backup_dest"
+          usage
+      fi
+  elif [ "${backup_dest:0:5}" = "s3://" ]; then
+      env -i $CREDENTIALS $xbcloud get ${backup_dest} xtrabackup_info 1>/dev/null
+  else
+      echo "Can't find $backup_dest backup"
+      usage
+  fi
+
+  if [ ! -d "$dest_dir" ]; then
+      printf "[ERROR] '%s' is not local directory.\n\n" "$dest_dir"
+      usage
+  fi
 
-    if [ "${backup_dest:0:4}" = "pvc/" ]; then
-        if ! $ctrl get "$backup_dest" 1>/dev/null; then
-            printf "[ERROR] '%s' PVC doesn't exists.\n\n" "$backup_dest"
-            usage
-        fi
-    elif [ "${backup_dest:0:5}" = "s3://" ]; then
-        env -i $CREDENTIALS $xbcloud get ${backup_dest} xtrabackup_info 1>/dev/null
-    else
-        usage
-    fi
-
-    if [ ! -d "$dest_dir" ]; then
-        printf "[ERROR] '%s' is not local directory.\n\n" "$dest_dir"
-        usage
-    fi
 }
 
 start_tmp_pod() {
     local backup_pvc=$1
 
     $ctrl delete pod/backup-access 2>/dev/null || :
-    cat - <<-EOF | $ctrl apply -f -
-		apiVersion: v1
-		kind: Pod
-		metadata:
-		  name: backup-access
-		spec:
-		      containers:
-		      - name: xtrabackup
-		        image: percona/percona-xtradb-cluster-operator:0.3.0-backup
-		        volumeMounts:
-		        - name: backup
-		          mountPath: /backup
-		      restartPolicy: Never
-		      volumes:
-		      - name: backup
-		        persistentVolumeClaim:
-		          claimName: ${backup_pvc#pvc/}
-	EOF
+
+    if [ -n "$namespace" ]; then
+      cat - <<-EOF | $ctrl apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: backup-access
+  namespace: $namespace
```

Comment:
It is not needed to specify `namespace: $namespace` inside YAML because it is already specified inside `$ctrl`.
In other words - it is not needed to change the `start_tmp_pod` function at all; it creates Pod in the right namespace because of `$ctrl` content.

Reply:
Thanks for review. That's done. Tested the script successfully:

```
$ ~/Documents/git/percona-xtradb-cluster-operator-fork/deploy/backup $ ./copy-backup.sh cron-mattermost-clust-fs-pvc-20210322000012-372f8 /tmp/`date +%Y-%m-%d_%H-%M-%S` mm-p
pvc/xb-cron-mattermost-clust-fs-pvc-20210322000012-372f8pod/backup-access created
Starting pod....[done]

Downloading started
tar: Removing leading `/' from member names
Downloading finished
pod "backup-access" deleted

You can recover data locally with following commands:
    $ service mysqld stop
    $ rm -rf /var/lib/mysql/*
    $ cat /tmp/2021-03-24_13-29-43/xtrabackup.stream | xbstream --decompress -x -C /var/lib/mysql
    $ xtrabackup --prepare --target-dir=/var/lib/mysql
    $ chown -R mysql:mysql /var/lib/mysql
    $ service mysqld start

$ ~/Documents/git/percona-xtradb-cluster-operator-fork/deploy/backup $ ls -l /tmp/2021-03-24_13-29-43
total 262200
-rw-r--r-- 1 mac wheel        52 Mar 24 13:29 md5sum.txt
-rw-r--r-- 1 mac wheel       106 Mar 24 13:29 sst_info
-rw-r--r-- 1 mac wheel 256477268 Mar 24 13:39 xtrabackup.stream
$ ~/Documents/git/percona-xtradb-cluster-operator-fork/deploy/backup $ k get pxc-backup -n mm-p
NAME                                                CLUSTER              STORAGE   DESTINATION                                                STATUS      COMPLETED   AGE
backup-mattermost-0001                              mattermost-cluster   fs-pvc    pvc/xb-backup-mattermost-0001                              Succeeded   3d12h       3d12h
backup-mattermost-0002                              mattermost-cluster   fs-pvc    pvc/xb-backup-mattermost-0002                              Succeeded   2d13h       2d13h
backup-mattermost-128371                            mattermost-cluster   fs-pvc    pvc/xb-backup-mattermost-128371                            Succeeded   5d14h       5d15h
cron-mattermost-clust-fs-pvc-20210322000012-372f8   mattermost-cluster   fs-pvc    pvc/xb-cron-mattermost-clust-fs-pvc-20210322000012-372f8   Succeeded   2d13h       2d13h
cron-mattermost-clust-fs-pvc-20210323000007-372f8   mattermost-cluster   fs-pvc    pvc/xb-cron-mattermost-clust-fs-pvc-20210323000007-372f8   Succeeded   12h         37h
cron-mattermost-clust-fs-pvc-20210324000014-372f8   mattermost-cluster   fs-pvc    pvc/xb-cron-mattermost-clust-fs-pvc-20210324000014-372f8   Succeeded   12h         13h
$ ~/Documents/git/percona-xtradb-cluster-operator-fork/deploy/backup $
```